app/src/test/java/com/codesoom/assignment/application/UserServiceTest.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    @Nested
+    @DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+    class createUser_메소드는 {
+
+        @Nested
+        @DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+        class 등록할_User가_주어진다면 {
+
+            UserData givenUserData;
+
+            @BeforeEach
+            void prepare() {
+                givenUserData = testUserDatas.get(0);
+            }
+
+            @Test
+            @DisplayName("User 객체를 생성한다")
+            void User를_생성하고_리턴한다() {
+                User user = userService.createUser(givenUserData);
+
+                User foundUser = userService.getUser(user.getId());
+
+                assertThat(foundUser.getName()).isEqualTo(givenUserData.getName());
+                assertThat(foundUser.getEmail()).isEqualTo(givenUserData.getEmail());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+    class updateUser_메소드는 {
+
+        @Nested
+        @DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+        class 등록된_User의_id와_수정할_User가_주어진다면 {
+
+            UserData givenUserData;
+            Long givenId;
+
+            @BeforeEach
+            void prepaer() {
+                User user = userService.createUser(testUserDatas.get(0));
+                givenId = user.getId();
+                givenUserData = testUserDatas.get(1);
+            }
+
+            @Test
+            void 해당_id의_User를_수정하고_리턴한다() {
+                userService.updateUser(givenId, givenUserData);
+
+                User foundUser = userService.getUser(givenId);
+
+                assertThat(foundUser).isNotNull();
+                assertThat(foundUser.getName()).isEqualTo(givenUserData.getName());
+            }
+        }
+
+        @Nested
+        @DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+        class 등록되지_User의_id와_수정할_User가_주어진다면 {
+
+            UserData givenUserData;
+            Long givenInvalidId = 9999L;
+
+            @BeforeEach
+            void prepaer() {
+                givenUserData = testUserDatas.get(1);
+            }
+
+            @Test
+            void 해당_id의_User를_수정하고_리턴한다() {
+                assertThatThrownBy(() -> userService.updateUser(givenInvalidId, givenUserData)).isInstanceOf(UserNotFoundException.class);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+    class deleteUser_메소드는 {
+
+        @Nested
+        @DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+        class 등록된_User의_id가_주어진다면 {
+
+            Long givenId;
+
+            @BeforeEach
+            void prepare() {
+                User user = userService.createUser(testUserDatas.get(0));
+                givenId = user.getId();
+            }
+
+            @Test
+            void 등록된_User를_삭제하고_빈값이_리턴한다() {
+                userService.deleteUser(givenId);
```

Comment:
`deleteUser`는 `void`를 리턴하죠? 그렇다면 이렇게 해 주시면 됩니다.
```suggestion
            void 등록된_User를_삭제한다() {
                userService.deleteUser(givenId);
```

Reply:
아하 !! 감사합니다.